vnext/ReactUWP/Utils/ResourceBrushUtils.cpp

```diff
@@ -0,0 +1,141 @@
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOffPointerOver, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOffPressed, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOffDisabled, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOn, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOnPointerOver, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOnPressed, brush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchStrokeOnDisabled, brush);
+}
+
+void UpdateToggleSwitchThumbResourceBrushes(
+    const winrt::ToggleSwitch &toggleSwitch,
+    const winrt::Brush thumbBrush) {
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOff, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOffPointerOver, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOffPressed, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOffDisabled, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOn, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOnPointerOver, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOnPressed, thumbBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchKnobFillOnDisabled, thumbBrush);
+}
+
+void UpdateToggleSwitchTrackResourceBrushes(
+  const winrt::ToggleSwitch& toggleSwitch,
+  const winrt::Brush onTrackBrush,
+  const winrt::Brush offTrackBrush) {
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOn, onTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOnPointerOver, onTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOnPressed, onTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOnDisabled, onTrackBrush);
+
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOff, offTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOffPointerOver, offTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOffPressed, offTrackBrush);
+  UpdateResourceBrush(toggleSwitch, c_toggleSwitchFillOffDisabled, offTrackBrush);
+}
+
+bool IsObjectATextControl(const winrt::DependencyObject &object) {
+  return object.try_as<winrt::TextBox>() != nullptr || object.try_as<winrt::PasswordBox>() != nullptr ||
+      object.try_as<winrt::RichEditBox>() != nullptr || object.try_as<winrt::AutoSuggestBox>() != nullptr;
+}
+
+void UpdateControlBackgroundResourceBrushes(
+    const winrt::Windows::UI::Xaml::FrameworkElement &element,
+    const winrt::Media::Brush brush) {
+  if (IsObjectATextControl(element)) {
+    UpdateTextControlBackgroundResourceBrushes(element, brush);
+  }
+}
+
+void UpdateControlForegroundResourceBrushes(
+    const winrt::Windows::UI::Xaml::DependencyObject object,
+    const winrt::Media::Brush brush) {
+  if (IsObjectATextControl(object)) {
+    UpdateTextControlForegroundResourceBrushes(object, brush);
```

Comment:
>object [](start = 47, length = 6)

Can you cast object here to FrameworkElement? Text Edit Control is always FrameworkElement, so we can eliminate need of TryUpdateResourceBrush.

Reply:
@ddalp good catch, updated and pushed another iteration.